### DISCUSSION TITLE
Wrapping

### DISCUSSION THREAD

**Discussion by flobadob** [DotSpatial/DotSpatial#1533] [General]
Any way to wrap data at the -180 / 180 longitudes? I need to allow polygon drawing so these areas of the world become problematic.  I'm using shape files for the coastlines and a raster for basic topography / bathymetry. Thanks.

**Comment by jany-tenaj** [comment]
Check out [DatelineCrossingMap ](https://github.com/DotSpatial/DotSpatial/blob/master/Source/Core/DotSpatial.Controls/DatelineCrossingMap.cs) and maybe use the search for Dateline. 

Otherwise @sindizzy might be able to help you. He works with those things, too.

**Reply by sindizzy** [reply]
So in the world of aviation we have all sorts of polylines and polygons that cross the dateline and it becomes v_e_r_y problematic.

- I believe I tried the DatelineCrossingMap and the polygons would still tear.
- You may also try creating your polygons with coordinates that are based on the 0 to 360 longitudes. I've had some success with that methodology.
- The last thing I have tried, and has worked pretty well for me, is to break up the polygon into a multi-**FEATURE** polygon. One feature for the east side and one feature for the west side. At this point you may be tempted to create one feature with multi-**PART**. That has not worked well for me as the GIS blends the two parts together if they share a boundary.  I have a fairly good thread going [here](https://github.com/DotSpatial/DotSpatial/issues/1407) with some code that I have been using for quite some time.
